api/api-secret-configs-v1/src/test/groovy/com/thoughtworks/go/apiv1/secretconfigs/representers/SecretConfigRepresenterTest.groovy

```diff
@@ -0,0 +1,251 @@
+
+class SecretConfigRepresenterTest {
+
+  @Nested
+  class toJSON {
+    SecretConfig secretConfig
+
+    @BeforeEach
+    void setup() {
+      secretConfig = new SecretConfig("id", "plugin-id")
+      secretConfig.getConfiguration().add(ConfigurationPropertyMother.create("key1", false, "value1"))
+      secretConfig.getConfiguration().add(ConfigurationPropertyMother.create("key2", "secret", "AES:lzcCuNSe4vUx+CsWgN11Uw==:YotExzWbFv5w/7/HmpYp3g=="))
+
+      secretConfig.setRules(new Rules([
+        new Allow("refer", "PipelineGroup", "DeployPipelines"),
+        new Allow("view", "Environment", "DeployEnvironment"),
+        new Deny("refer", "PipelineGroup", "TestPipelines"),
+        new Deny("view", "Environment", "TestEnvironment")
+      ]))
+    }
+
+    @Test
+    void shouldSerializeToJSON() {
+      def json = toObjectString({ toJSON(it, secretConfig) })
+
+      JsonFluentAssert.assertThatJson(json).isEqualTo([
+        "_links"    : [
+          "doc" : [
+            "href": "https://api.gocd.org/19.3.0/#secret_configs"
+          ],
+          "find": [
+            "href": "http://test.host/go/api/admin/secret_configs/:config_id"
+          ],
+          "self": [
+            "href": "http://test.host/go/api/admin/secret_configs/id"
+          ]
+        ],
+        "id"        : "id",
+        "plugin_id" : "plugin-id",
+        "properties": [
+          [
+            "key"  : "key1",
+            "value": "value1"
+          ],
+          [
+            "key"            : "key2",
+            "encrypted_value": "AES:lzcCuNSe4vUx+CsWgN11Uw==:YotExzWbFv5w/7/HmpYp3g=="
+          ]
+        ],
+        "rules"     : [
+
+          [
+            "directive": "allow",
+            "action"   : "refer",
+            "resource" : "DeployPipelines",
+            "type"     : "PipelineGroup"
+          ],
+          [
+            "directive": "allow",
+            "action"   : "view",
+            "resource" : "DeployEnvironment",
+            "type"     : "Environment"
+          ],
+          [
+            "directive": "deny",
+            "action"   : "refer",
+            "resource" : "TestPipelines",
+            "type"     : "PipelineGroup"
+          ],
+          [
+            "directive": "deny",
+            "action"   : "view",
+            "resource" : "TestEnvironment",
+            "type"     : "Environment"
+          ]
+        ]
+      ])
+    }
+
+    @Test
+    void shouldSerializeEmptyObject() {
+      secretConfig = null
+      def json = toObjectString({ toJSON(it, secretConfig) })
+
+      JsonFluentAssert.assertThatJson(json).isEqualTo("{}")
+    }
+
+    @Test
+    void shouldNotRenderRulesIfEmpty() {
+      secretConfig.setRules(null)
+      def json = toObjectString({ toJSON(it, secretConfig) })
+
+      JsonFluentAssert.assertThatJson(json).isEqualTo([
+        "_links"    : [
+          "doc" : [
+            "href": "https://api.gocd.org/19.3.0/#secret_configs"
```

Comment:
Same as above

Reply:
Using `apiDocsUrl('#secret-configs')` to calculate documentation url.